libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -974,6 +974,14 @@ void CompilerUtils::convertType(
 				}
 				else
 				{
+					if (auto baseType = dynamic_cast<ArrayType const*>(typeOnStack.baseType()))
+						solUnimplementedAssert(
+							typeOnStack.isDynamicallySized() ||
+							!typeOnStack.isDynamicallyEncoded() ||
+							!baseType->isByteArray(),
```

Comment:
Why is this assertion different than the other?

Reply:
I used the test cases to tailor the asserts to them

Reply:
I basically lacked the understanding of the bigger picture.

Reply:
I am now extending the assert